corehq/apps/analytics/tasks.py

```diff
@@ -38,28 +38,19 @@
 HUBSPOT_EXISTING_USER_INVITE_FORM = "7533717e-3095-4072-85ff-96b139bcb147"
 HUBSPOT_COOKIE = 'hubspotutk'
 
-ANALYTICS_RETRIES = 3
-ANALYTICS_SLEEP = 5
 
-
-def _persistent_analytics_post(func, retries=ANALYTICS_RETRIES, sleep=ANALYTICS_SLEEP):
-    '''
-    this function can block for up to 15 seconds. please only call it from a async task so as not to block
-    page loads for that long.
-    '''
-    for i in range(retries):
-        try:
-            return func()
-        except requests.exceptions.HTTPError as e:
-            # if its a bad request, raise the exception because it is our fault
-            res = e.response
-            status_code = res.status_code if isinstance(res, requests.models.Response) else res.status
-            if status_code == 400:
-                raise
-            if i < retries - 1:
-                time.sleep(sleep)
-            else:
-                raise
+@task(bind=True, queue='background_queue', acks_late=True, ignore_result=True, default_retry_delay=10)
+def _persistent_analytics_post(self, func):
```

Comment:
I'm generally not a big fan of passing closures to celery since I think it can be problematic and also hard to debug.